htpasswd/auth.go

```diff
@@ -94,17 +108,21 @@ func (a *Auth) Authenticate(username, password string) (bool, error) {
 
 // Reload htpasswd file if it changed during current run
 func reloadIfChanged(a *Auth) {
-	fileInfo, err := os.Stat(a.filename)
-	if err != nil {
-		//On error, retain current file
-		return
+	parse := false
+	for _, file := range a.filenames {
+		fileInfo, err := os.Stat(file.name)
+		if err != nil {
+			//On error, retain current file
+			break
+		}
+		currentmodTime := fileInfo.ModTime()
+		if currentmodTime != file.modTime {
+			file.modTime = currentmodTime
+			parse = true
```

Comment:
No need for the variable, better:

    if currentmodTime != a.modTime {
        a.parse(a.filenames)
        return nil
    }